javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,16 @@
+package java.util.concurrent.locks
+
+class ReentrantLock(private[this] val fair: Boolean) {
+
+  private var locked = false
```

Comment:
Using a boolean `locked` is not correct, because of this chain of operations:
```scala
l.lock()
l.lock()
l.unlock()
expect(l.isLocked).toBeTruthy
l.unlock()
expect(l.isLocked).toBeFalsy
```